css/selectors/not.json

```diff
@@ -49,6 +49,55 @@
             "deprecated": false
           }
         },
+        "complex_selectors": {
```

Comment:
How is this different from the `selector_list` feature below? I'm not sure I understand the nuance between selector lists, compound selectors, and complex selectors. Can we do anything to make the distinction clearer (because it's not clear to me)?

Reply:
@rachelandrew Do you plan to return to this PR at some point?

Reply:
@ddbeck sorry I'd missed this request, I'll have a look tomorrow.

Reply:
@ddbeck there are examples on the documentation page at: https://developer.mozilla.org/en-US/docs/Web/CSS/:not the selector list being the penultimate, and a complex selector the last item.

I'm unclear as to what more you expect to be part of the BCD though.

Reply:
OK, I think I get the selector list distinction, but not the compound and complex distinction.

[The spec](https://www.w3.org/TR/selectors-4/#complex) defines a complex selector as "one or more compound selectors separated by combinators". By that definition, the last item in the linked examples demonstrates a compound (not complex) selector inside the `:not`. Have I badly misunderstood this PR to mean that the new feature describes whether `:not` itself—not its contents—can be part of a complex selector?

If this is, in fact, about the contents of `:not` then I'd like to know what not supporting this feature means. For example, if support is `false`, does that mean _neither_ compound nor complex selectors work (e.g. `:not(span)` works, but neither `:not(span.a)` nor `:not(span .b)` work)? Or Is it possible for a browser to support compound but not complex selectors (e.g. `:not(span)` and `:not(span.a)` work, but `:not(span .b)` doesn't)?

Reply:
(sorry, forgot to tag you on this @rachelandrew)

Reply:
@ddbeck I keep coming back to try and reply to this, but I'm struggling to understand the problem you are having.

Do you think the MDN page needs updating? As far as support goes I think complex selectors are all pretty much lumped together. 

I don't really understand what you are driving at with,

> Have I badly misunderstood this PR to mean that the new feature describes whether :not itself—not its contents—can be part of a complex selector?

We're talking about what matches for `:not`. Essentially what was added in Selectors 4: https://drafts.csswg.org/selectors-4/#negation (see the note)

Reply:
I think I misunderstood one part from the outset, then figured it out, and muddied things a little. Let me try to reboot things here a bit.

Here's the problem I have right now: I think the description of this feature creates ambiguities. I don't know your intent, so I can't tell you how to fix those ambiguities. Once I know more about what you mean by this data, then I can suggest some paths forward.

Here are the ambiguities I'm trying to fix:

1. The other subfeature of `:not()` is described as "Selector list argument". This feature doesn't mention the word argument. Is this feature correctly understood to also be about the argument to `:not()`? That is, is it correct to read this subfeature as "Complex and compound selector argument"?

   (I'm pretty sure it is, but I want to double check to be sure.)
   
1. This subfeature refers to selectors plural, while the other subfeature refers to selector list support (that is, whether `:not` accepts more than one selector as an argument). Is it correct to understand this subfeature to mean "`:not` accepts one complex or compound selector as an argument"? Or am I supposed to understand this as subsuming the selector list question too?

   (Again, I'm pretty sure it means that it accepts one complex/compound selector, not a selector list, but the plural makes me want to double check.)
   
2. "Complex and compound selectors" sounds like two things (e.g., "Complex selector argument" and "Compound selector argument"). As I see it, a `false` value for this feature, as it stands, could be plausibly read by web developers as:

   - "Supports neither a complex nor a compound selector as an argument to `:not()`"
   - "Supports a compound selector but not a complex selector as an argument to `:not()`"
   - "Supports a compound or complex selector as an argument to `:not()`, but I haven't told you which one"
   - "Supports a complex but not a compound selector as an argument to `:not()`"

   (I happen to know the definition of complex better now than I did at the start, so I know the last two are definitionally impossible. But I thought this might be the case when we started! And I'm still not sure about the first two bullets.)

I hope this helps explain what I'm getting at here. 😅